### PR TITLE
Add subctl show connections cmd

### PR DESCRIPTION
**Pull request by maayanf24**
Resolves issue #469

Signed-off-by: Maayan Friedman <maafried@redhat.com>

### CODE REVIEW

File: pkg/subctl/cmd/show_connections.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+func newConnectionsStatusFrom(gateway string, cluster string, remoteIp string, cableDriver string, subnets string, status submv1.ConnectionStatus) connectionsStatus {
+	v := connectionsStatus{
+		gateway:     gateway,
+		cluster:     cluster,
+		remoteIp:    remoteIp,
+		cableDriver: cableDriver,
+		subnets:     subnets,
+		status:      status,
+	}
+	return v
```
Comment (tpantelis) [inline pkg/subctl/cmd/show_connections.go:32]:
```suggestion
	return connectionsStatus{
		gateway:     gateway,
		cluster:     cluster,
		remoteIp:    remoteIp,
		cableDriver: cableDriver,
		subnets:     subnets,
		status:      status,
	}
```

File: pkg/subctl/cmd/show_connections.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+var showConnectionsCmd = &cobra.Command{
+	Use:   "connections",
+	Short: "Get information on your connections related to submariner",
```
Comment (tpantelis) [inline pkg/subctl/cmd/show_connections.go:37]:
```suggestion
	Short: "Show connections with other clusters",
```

File: pkg/subctl/cmd/show_connections.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	Long:  `This command shows the status of submariner connections in your cluster.`,
```
Comment (tpantelis) [inline pkg/subctl/cmd/show_connections.go:38]:
```suggestion
	Long:  `This command shows information about submariner endpoint connections with other cluster.`,
```

File: pkg/subctl/cmd/show_connections.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	exitOnError("Unable to get the Submariner client", err)
+
+	existingCfg, err := submarinerClient.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, v1.GetOptions{})
+	if err != nil {
+		exitOnError("error reading from submariner client", err)
+	}
+
+	gateways := existingCfg.Status.Gateways
+	if gateways == nil {
+		exitWithErrorMsg("no gateways found")
```
Comment (tpantelis) [inline pkg/subctl/cmd/show_connections.go:60]:
```suggestion
		exitWithErrorMsg("No endpoints found")
```

File: pkg/subctl/cmd/show_connections.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+func showConnections(cmd *cobra.Command, args []string) {
+	var status []connectionsStatus
+	status = getConnectionsStatus(status)
+	printConnections(status)
+}
+
+func printConnections(connections []connectionsStatus) {
+	template := "%-20s%-16s%-16s%-24s%-40s%-16s\n"
+
+	fmt.Printf(template, "GATEWAY", "CLUSTER", "REMOTE_IP", "CABLE-DRIVER", "SUBNETS", "STATUS")
```
Comment (tpantelis) [inline pkg/subctl/cmd/show_connections.go:88]:
```suggestion
	fmt.Printf(template, "GATEWAY", "CLUSTER", "REMOTE IP", "CABLE DRIVER", "SUBNETS", "STATUS")
```

File: pkg/subctl/cmd/show_connections.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+	"strings"
+
+	"github.com/spf13/cobra"
+	submarinerclientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	submv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type connectionsStatus struct {
```
Comment (tpantelis) [inline pkg/subctl/cmd/show_connections.go:14]:
```suggestion
type connectionStatus struct {
```

**Comment by tpantelis** [comment]
@maayanf24 When addressing review comments, we generally commit a new patch rather than amending as it helps reviewers to see what actually changed. Unless it's a small patch and/or the changes were small.

**Comment by maayanf24** [comment]
> @maayanf24 When addressing review comments, we generally commit a new patch rather than amending as it helps reviewers to see what actually changed. Unless it's a small patch and/or the changes were small.

@tpantelis  Thanks for clarifying this

File: pkg/subctl/cmd/show_connections.go [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+		cableDriver: cableDriver,
+		subnets:     subnets,
+		status:      status,
+	}
+}
+
+var showConnectionsCmd = &cobra.Command{
+	Use:   "connections",
+	Short: "Show cluster connectivity information",
+	Long:  `This command shows information about submariner endpoint connections with other cluster.`,
```
Comment (tpantelis) [inline pkg/subctl/cmd/show_connections.go:37]:
```suggestion
	Long:  `This command shows information about submariner endpoint connections with other clusters.`,
```

File: pkg/subctl/cmd/show_connections.go [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+type connectionStatus struct {
+	gateway     string
+	cluster     string
+	remoteIp    string
+	cableDriver string
+	subnets     string
+	status      submv1.ConnectionStatus
+}
+
+func newConnectionsStatusFrom(gateway string, cluster string, remoteIp string, cableDriver string, subnets string, status submv1.ConnectionStatus) connectionStatus {
```
Comment (mangelajo) [inline pkg/subctl/cmd/show_connections.go:23]: Same comment as in the other patch, it's cleaner to create the structure right away, unless you had some business logic for creating the structure.

File: pkg/subctl/cmd/show_connections.go [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+	Short: "Show cluster connectivity information",
+	Long:  `This command shows information about submariner endpoint connections with other clusters.`,
+	Run:   showConnections,
+}
+
+func init() {
+	showCmd.AddCommand(showConnectionsCmd)
+}
+
+func getConnectionsStatus(status []connectionStatus) []connectionStatus {
```
Comment (mangelajo) [inline pkg/subctl/cmd/show_connections.go:45]:
```suggestion
func getConnectionsStatus() []connectionStatus {
	var status []connectionStatus
```

File: pkg/subctl/cmd/show_connections.go [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+				connection.Status))
+		}
+	}
+
+	return status
+}
+
+func showConnections(cmd *cobra.Command, args []string) {
+	var status []connectionStatus
+	status = getConnectionsStatus(status)
```
Comment (mangelajo) [inline pkg/subctl/cmd/show_connections.go:80]:
```suggestion
	status = getConnectionsStatus()
```

**Pull request merged**